distuv/bernoulli.go

```diff
@@ -0,0 +1,132 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package distuv
+
+import (
+	"math"
+	"math/rand"
+)
+
+// Bernoulli represents a random variable whose value is 1 with probability p and
+// value of zero with probability 1-P. The value of P must be between 0 and 1.
+// More information at https://en.wikipedia.org/wiki/Bernoulli_distribution.
+type Bernoulli struct {
+	P      float64
+	Source *rand.Rand
+}
+
+// CDF computes the value of the cumulative density function at x.
+func (b Bernoulli) CDF(x float64) float64 {
+	if x < 0 {
+		return 0
+	}
+	if x < 1 {
+		return 1 - b.P
+	}
+	return 1
+}
+
+// Entropy returns the entropy of the distribution.
+func (b Bernoulli) Entropy() float64 {
+	if b.P == 0 {
+		return 0
+	}
+	if b.P == 1 {
+		return 1
+	}
+	q := 1 - b.P
+	return -b.P*math.Log(b.P) - q*math.Log(q)
+}
+
+// ExKurtosis returns the excess kurtosis of the distribution.
+func (b Bernoulli) ExKurtosis() float64 {
+	pq := b.P * (1 - b.P)
+	return (1 - 6*pq) / pq
+}
+
+// LogProb computes the natural logarithm of the value of the probability density function at x.
+func (b Bernoulli) LogProb(x float64) float64 {
+	if x == 0 {
+		return math.Log(1 - b.P)
+	}
+	if x == 1 {
+		return math.Log(b.P)
+	}
+	return math.Inf(-1)
+}
+
+// Mean returns the mean of the probability distribution.
+func (b Bernoulli) Mean() float64 {
+	return b.P
+}
+
+// Median returns the median of the probability distribution.
+func (b Bernoulli) Median() float64 {
+	p := b.P
+	q := 1 - p
+	switch {
+	case p < q:
```

Comment:
I think that `q` is not necessary. These comparisons are the same as comparing `p` to 0.5. Could you add it to #108 ?